### PR TITLE
Add natural sorting

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
This hopefully helps to fix TimSort errors.
With previous implementation transient comparison was not well defined:
a < b
b < c
--> a < c

This should now be the case.
I also added natural sorting:
1, 01, 001, 2, 002, 11, 12, 100

- first use value without leading zero
- if then value is equal, sort by leading zeros

### CODE REVIEW

File: src/main/java/com/owncloud/android/datamodel/OCFile.java [hunk 1]
```diff
@@ -618,10 +618,8 @@ public int describeContents() {
     }
 
     @Override
-    public int compareTo(OCFile another) {
-        if (isFolder() && another.isFolder()) {
-            return getRemotePath().toLowerCase(Locale.ROOT).compareTo(another.getRemotePath().toLowerCase(Locale.ROOT));
-        } else if (isFolder()) {
+    public int compareTo(@NonNull OCFile another) {
+        if (isFolder()) {
```
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/datamodel/OCFile.java]: This will now fail when both are folders! because a-b = -1 and b-a = -1 (!)

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/datamodel/OCFile.java]: Right. I changed this and added more testcases.

**Comment by AndyScherzinger** [comment]
:+1:

**Comment by mario** [comment]
👍

**Pull request merged**